src/cpp/fastdds/xtypes/utils.cpp

```diff
@@ -0,0 +1,1015 @@
+        {
+            long double value;
+            ReturnCode_t ret = data->get_float128_value(value, member_id);
+            if (RETCODE_OK == ret)
+            {
+                json_insert(member_name, value, output);
+            }
+            else
+            {
+                EPROSIMA_LOG_WARNING(XTYPES_UTILS, "Error encountered while serializing TK_FLOAT128 member to JSON.");
+            }
+            return ret;
+        }
+        case TK_CHAR8:
+        {
+            char value;
+            ReturnCode_t ret = data->get_char8_value(value, member_id);
+            if (RETCODE_OK == ret)
+            {
+                std::string aux_string_value({value});
+                json_insert(member_name, aux_string_value, output);
+            }
+            else
+            {
+                EPROSIMA_LOG_WARNING(XTYPES_UTILS, "Error encountered while serializing TK_CHAR8 member to JSON.");
+            }
+            return ret;
+        }
+        case TK_CHAR16:
+        {
+            wchar_t value;
+            ReturnCode_t ret = data->get_char16_value(value, member_id);
+            if (RETCODE_OK == ret)
+            {
+                // Insert UTF-8 converted value
+                std::wstring aux_wstring_value({value});
+                std::wstring_convert<std::codecvt_utf8<wchar_t>> converter;
+                std::string utf8_value = converter.to_bytes(aux_wstring_value);
+                json_insert(member_name, utf8_value, output);
+            }
+            else
+            {
+                EPROSIMA_LOG_WARNING(XTYPES_UTILS, "Error encountered while serializing TK_CHAR16 member to JSON.");
+            }
+            return ret;
+        }
+        case TK_STRING8:
+        {
+            std::string value;
+            ReturnCode_t ret = data->get_string_value(value, member_id);
+            if (RETCODE_OK == ret)
+            {
+                json_insert(member_name, value, output);
+            }
+            else
+            {
+                EPROSIMA_LOG_WARNING(XTYPES_UTILS, "Error encountered while serializing TK_STRING8 member to JSON.");
+            }
+            return ret;
+        }
+        case TK_STRING16:
+        {
+            std::wstring value;
+            ReturnCode_t ret = data->get_wstring_value(value, member_id);
+            if (RETCODE_OK == ret)
+            {
+                // Insert UTF-8 converted value
+                std::wstring_convert<std::codecvt_utf8<wchar_t>> converter;
+                std::string utf8_value = converter.to_bytes(value);
+                json_insert(member_name, utf8_value, output);
+            }
+            else
+            {
+                EPROSIMA_LOG_WARNING(XTYPES_UTILS, "Error encountered while serializing TK_STRING16 member to JSON.");
+            }
+            return ret;
+        }
+        case TK_ENUM:
+        {
+            int32_t value;
+            ReturnCode_t ret = data->get_int32_value(value, member_id);
```

Comment:
Test if this would work with smaller precision. In principle it should work as type promotion would be applied.